napari/_qt/qt_main_window.py

```diff
@@ -434,6 +437,8 @@ def closeEvent(self, event):
                 time.sleep(0.1)
                 QApplication.processEvents()
 
+        self._qt_viewer.dims.stop()
```

Comment:
Is there a reason we can't rely on `QtViewer.closeEvent` from calling `QtDims.stop`? It's not ideal that we have to call this line in two different places to handle the two different close paths in `QtMainWindow`.